Source/CarthageKit/Project.swift

```diff
@@ -13,13 +13,33 @@ import ReactiveCocoa
 /// Carthage’s bundle identifier.
 public let CarthageKitBundleIdentifier = NSBundle(forClass: Project.self).bundleIdentifier!
 
+/// The fallback dependencies URL to be used in case
+/// the intended ~/Library/Caches/org.carthage.CarthageKit cannot
+/// be found or created.
+private let fallbackDependenciesURL: NSURL = {
+	let homePath: String
+	if let homeEnvValue = NSProcessInfo.processInfo().environment["HOME"] as? NSString {
+		homePath = homeEnvValue.stringByAppendingPathComponent(".carthage")
+	} else {
+		homePath = "~/.carthage".stringByExpandingTildeInPath
+	}
+	return NSURL.fileURLWithPath(homePath, isDirectory:true)!
+}()
+
 /// ~/Library/Caches/org.carthage.CarthageKit/
 private let CarthageUserCachesURL: NSURL = {
+	let fileManager = NSFileManager.defaultManager()
+	
 	let URL: Result<NSURL, NSError> = try({ (error: NSErrorPointer) -> NSURL? in
-		NSFileManager.defaultManager().URLForDirectory(NSSearchPathDirectory.CachesDirectory, inDomain: NSSearchPathDomainMask.UserDomainMask, appropriateForURL: nil, create: true, error: error)
-	})
-
-	let fallbackDependenciesURL = NSURL.fileURLWithPath("~/.carthage".stringByExpandingTildeInPath, isDirectory:true)!
+		fileManager.URLForDirectory(NSSearchPathDirectory.CachesDirectory, inDomain: NSSearchPathDomainMask.UserDomainMask, appropriateForURL: nil, create: true, error: error)
+	}).flatMap { cachesURL in
+		if fileManager.isWritableFileAtPath(cachesURL.absoluteString!) {
```

Comment:
I think this should be checking the Carthage-specific directory of the caches directory. So this line:

```swift
let dependenciesURL = URL.value?.URLByAppendingPathComponent(CarthageKitBundleIdentifier, isDirectory: true)
```

should already have been executed.